packages/astro/src/core/cookies/cookies.ts

```diff
@@ -17,7 +17,7 @@ export interface AstroCookieGetOptions {
 	decode?: (value: string) => string;
 }
 
-type AstroCookieDeleteOptions = Pick<AstroCookieSetOptions, 'domain' | 'path'>;
+type AstroCookieDeleteOptions = Omit<CookieSerializeOptions, 'expires' | 'maxAge'>;
```

Comment:
We intentionally don't just support all options from `cookie` so that we know what we support and don't start supporting something accidentally because `cookie` adds new options in a minor. So I'd rather we not do this and instead explicitly add the new options we mean to support.

Reply:
I was under the impression that our semver range would prevent that.

Reply:
@matthewp would swapping `CookieSerializeOptions` with `AstroCookieSetOptions` address your concern?

Also, do you think it makes sense to construct `AstroCookieSetOptions` using `Pick<CookieSerializeOptions, ...>` instead of manually rewriting it? Currently, it’s missing the IDE annotations form the `cookie` package which is nice to have.

Reply:
Yes it would. Fine with using Pick as well, as long as we limit the options to what we knowingly support. Thanks!

Reply:
Taken care of.